pyccel/parser/semantic.py

```diff
@@ -1701,7 +1624,7 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, **settings):
                     d_lhs['allocatable'] = False
                     d_lhs['is_pointer' ] = True
 
-                    rhs = self.update_variable(rhs, is_target=True)
```

Comment:
shouldn't we keep `update_variable` and make the changes there instead of doing it directly
which will make it easier to debug.

Reply:
So replace
`rhs.is_target=True`
with `rhs = self.update_variable(rhs, is_target=True)`
and have:
```
def update_variable(self, var, **kwargs):
    for a, v in kwargs.items():
        assert(hasattr(var, a))
        setattr(var, a, v)
```
I'm not sure how this is easier to debug

Reply:
I think it's safer if all changes were done by this method.
 it's also easier to debug because we know it was done by this method and we only need to print the kwargs to see what was changed.
the other approach makes it harder to find the lines where we made the changes

Reply:
I see your point about debugging, although I don't know if we should be structuring our code around debugging. We can just as easily put a print in the setters and this will reduce the debugging noise as if you have a problem with `allocatable` you probably don't care about `is_target` for example.

Can you explain why it's safer?

Reply:
@saidctb I don't see a good reason to have an `update_variable` function to perform an operation that in Python only requires an assignment... If you want to make the code clearer, add comments, docstring and documentation to it! Not useless functions.

With regard to safety, the new attribute setters perform type checks that were not performed before, so using them is safer than using `update_variable`.